src/lib/BindToStores.react.js

```diff
@@ -1,47 +1,121 @@
 import React from 'react'
-import StorePrototype from './StorePrototype'
 
-// let's say for now, stores is a dictionary of name and object
-export default function bindResources(Component, resourceName) {
-  function extractStores() { return Object.getOwnPropertyNames(stores).map(store => stores[store]) }
-  function capitalize(word) { return word.charAt(0).toUpperCase() + word.slice(1) }
-  function singularize(word) { return (word.slice(-1) === 's' ? word.slice(0, -1) : word) }
+import Container from '../lib/Container'
+import storePrototype from './StorePrototype'
 
-  var bootAction = `fetch${capitalize(resourceName)}`
+export default function bindResources(Component, resources, onBoundUpdate = null, resourceId = null) {
+  const stores = {}
 
-  // figure out how to pass in actions
-  // import actionClass from `PATH TO ACTIONS/${singularize(capitalize(resourceName))}Actions`
+  function capitalize(word, pascal = true) {
+    return (pascal ? word.charAt(0).toUpperCase() : word.charAt(0)) + word.slice(1).replace(/(\_\w)/g, match => match[1].toUpperCase())
+  }
+  function storeNames() { return Object.getOwnPropertyNames(stores) }
 
-  var actionType = actionClass.Types[`GOT_${resourceName.toUpperCase()}`]
-  var resourceStore = StorePrototype(actionType)
-  var stores = {}
-  stores[resourceName] = resourceStore
+  var resourceConfigs = (Array.isArray(resources) ? resources : [resources])  // Handle old, one-resource pattern too.
+
+  var bootObjects = []  // Objects containing all necessary information for firing the bootAction of each resource
+    // and the tracking of the loading state of each resource.
+    // Example of the bootObject for an 'item' resource:
+       // {bootAction: 'fetchClient', actionClass: 'ClientActions', resourceType: 'item', resourceId: '12'}
+
+  // Construct a bootObject for every resource:
+  resourceConfigs.forEach(resourceConfig => {
+    const resourceType = Object.getOwnPropertyNames(resourceConfig)[0]
+    const resourceName = resourceConfig[resourceType]
+
+    // Get the action class--previously registered with the Container--and action Type used to fetch the resource data:
+    const actionClass = Container.getAction(resourceName)
+    const actionType = actionClass.Types[`GOT_${resourceName.toUpperCase()}`]
+
+    // Get or create the store used to hold the resource data:
+    var resourceStore = Container.getStore(resourceName)
+    if (!resourceStore)
+      resourceStore = Container.registerStore(resourceName, storePrototype(actionType))
+
+    // Add the store to the list of store the BoundComponent will listen to:
+    stores[resourceName] = resourceStore
+
+    bootObjects.push({
+      actionClass,
+      bootAction: `fetch${capitalize(resourceName)}`,
+      loadingName: `loading${capitalize(resourceName)}`,
+      resourceId: (resourceId ? resourceId : `${capitalize(resourceName, false)}Id`),
+      resourceType
+    })
+  })
 
   const BoundComponent = React.createClass({
```

Comment:
Chose to leave it as ES5 for now since this PR is already larger than I would have liked.